File: app/src/main/java/com/cartenz/cartenztaxcore/feature/login/LoginPresenter.java


package com.cartenz.cartenztaxcore.feature.login;

import android.content.Context;
import android.support.annotation.Nullable;

import com.cartenz.cartenztaxcore.App;
import com.cartenz.cartenztaxcore.api.dao.LoginDao;
import com.cartenz.cartenztaxcore.api.repository.LoginRepository;
import com.cartenz.core.api.MySubscriber;
import com.cartenz.core.base.BaseSubscriber;

import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

final class LoginPresenter extends BaseSubscriber implements LoginContract.PresenterInterface {


    @Nullable
    private LoginContract.View mLoginView;
    private Context context;


    @Override
    public void setView(Context context, LoginContract.View view) {
        this.context = context;
        this.mLoginView = view;

    }


    @Override
    public void callLogin(String username, String password) {
        LoginRepository repo = new LoginRepository(App.getApi(), username, password);
        addSubscription(repo.post().subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new MySubscriber<LoginDao>() {
                    @Override
                    public void onComplete() {

                    }

                    @Override
                    public void onError(String message, String errorMessage) {
                        mLoginView.loginResult(errorMessage);
                    }

                    @Override
                    public void onSuccess(LoginDao loginDao) {
                        mLoginView.loginResult(null);
                    }

                }));
    }

    @Override
    public void dropView() {
        finishSubscriber();
        mLoginView = null;
    }


}

File: kotlinapp/src/main/java/com/cartenz/kotlinapp/feature/login/LoginPresenter.kt
package com.cartenz.kotlinapp.feature.login

import android.content.Context
import com.cartenz.kotlin_core.BaseSubscriber
import com.cartenz.kotlin_core.api.MySubscriber
import com.cartenz.kotlinapp.api.dao.LoginDao
import com.cartenz.kotlinapp.api.repository.LoginRepository
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

internal class LoginPresenter : BaseSubscriber(), LoginContract.PresenterInterface {

    private var mLoginView: LoginContract.View? = null
    private var context: Context? = null

    override fun setView(context: Context, view: LoginContract.View) {
        this.context = context
        this.mLoginView = view
    }

    override fun callLogin(username: String, password: String) {
        val repo = LoginRepository(username, password)
        addSubscription(repo.post().subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object : MySubscriber<LoginDao>() {
                    override fun onComplete() {

                    }

                    override fun onError(message: String?, errorMessage: String?) {
                        mLoginView!!.loginResult(errorMessage!!)
                    }

                    override fun onSuccess(loginDao: LoginDao) {
                        mLoginView!!.loginResult("")
                    }

                }))
    }

    override fun dropView() {
        finishSubscriber()
        mLoginView = null
    }

}
